crates/fluvio-protocol/src/core/wasm_bytes.rs

```diff
@@ -0,0 +1,87 @@
+use std::io::Error;
+use std::io::ErrorKind;
+
+use bytes::{Bytes, BufMut};
+
+use crate::{Encoder, Version};
+
+/// Represnts a SmartModule WASM File bytes.
+///
+/// Provides a `Encoder` implementation optimized for WASM files used in
+/// SmartModules.
+pub struct WasmBytes(Vec<u8>);
+
+impl From<Vec<u8>> for WasmBytes {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self(bytes)
+    }
+}
+
+impl Encoder for WasmBytes {
+    fn write_size(&self, version: Version) -> usize {
+        self.0
+            .iter()
+            .fold(4, |sum, val| sum + val.write_size(version))
```

Comment:
this is still inefficient: `val.write_size` since it will recursively calculate.  But since we it know it's byte, we can calculate length